khal/controllers.py

```diff
@@ -96,6 +96,11 @@ def get_agenda(collection, locale, dates=None, firstweekday=0, days=None, events
             logging.fatal(error)
             sys.exit(1)
 
+    if week:
+        dates = [d - datetime.timedelta((d.weekday() - firstweekday) % 7)
```

Comment:
this probably should be `timedelta` instead of `datetime.timedelta` (we don't import datetime but do `from datetime import datetime, date, timedelta`)